Netable/Netable/Netable.swift

```diff
@@ -125,7 +125,14 @@ open class Netable {
                 switch decoded {
                 case .success(let raw):
                     let finalizedData = request.finalize(raw: raw)
-                    self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedData))
+
+                    switch finalizedData {
+                    case .success(let finalizedResult):
+                        self.log(.requestSuccess(request: requestInfo, taskTime: time, statusCode: response.statusCode, responseData: data, finalizedResult: finalizedResult))
+                    case .failure(let error):
+                        self.log(.requestFailed(request: requestInfo, taskTime: time, error: error))
```

Comment:
I know it's not new to this change (because previously it was not handling those errors at all and just passing any error straight to the completion handler), but while we are tweaking this, should this maybe be a `throw error` instead? Right now the semantics of failure here are subtly different that other decoding errors, because it doesn't go to the regular error handling path these errors CAN'T be retried. Although I think we could argue that decoding errors are of a slightly different character than network errors and maybe shouldn't be retried as much (since you are unlikely to get different results) we DO currently have the ability to retry JSON decoding errors, and it seems like we should at least be consistent between JSON decoding and finalization errors, since they are pretty much the same thing.

Reply:
Oh great question... I think it's a little strange to allow for retries since I can't really come up with a plausible scenario where you'd expect the retry to succeed but I think it's slightly more important to be consistent, so lets go with that.

Reply:
Yeah, I'm kind in the same boat, retrying decoding errors seems not super useful in practice, but being inconsistent between JSON and finalization errors seems worse, and it being impossible for the user to decide to retry those errors also is not great.

We should probably consider having some default set of errors that are not retryable (including JSON decoding errors and these), as long as there was some way to turn that off. Maybe there's a default implementation of allowRetry that get called if you don't specify one that suppresses both of them by default, but if you implement your own you've go the option to allow them?

That would also potentially solve the crumminess with timeout errors. The potential consequences of retying those (default timeouts are long enough that allowing retries could make a request take minutes) was bad enough that I decided to explicitly suppress those when implementing retries. However, It WOULD be safe, and possibly desirable, to retry timeouts if you knew that you had a relatively short timeout window (or were interacting with a server that you know timed out a lot), but there's no way to turn them on. 

I'll file a bug for that.